### PR TITLE
SRD CQ notifications

### PR DESCRIPTION
**Pull request by gal-pressman**
Add support for SRD CQ notifications.
The first patch fixes a bug where the same completion channel is used for both TX and RX and messes up the test.

### CODE REVIEW

**Comment by amitgeron** [comment]
Hi,

Can you please add more details what issue the first patch actually resolves?

Thanks

**Comment by galpress** [comment]
IIRC, the test uses the same completion channel and does:

- Send packet
- Remote side receives the packet and send another packet
- Receive packet from remote

And assumes that the next event is necessarily from the send completion, but it could theoretically be from the receive completion.

@firasj might remember the exact details.

**Comment by amitgeron** [comment]
The unidirectional BW test for the client (run_iter_bw) is not supposed to receive data at all, so an event should never be delivered for the recv_cq, and the server side (run_iter_bw_server) isn't supposed to send any data, so an event should never be delivered for the send_cq.

Did you encounter any bug or issues when running the tests?

**Comment by galpress** [comment]
It was a real issue, some tests (send_lat for example) do both send and receive, and they might fail without this fix.

**Comment by mrgolin** [comment]
@HassanKhadour Can you please merge this PR or suggest some extra improvements?
Thanks!

**Comment by sshaulnv** [comment]
@firasj could you please provide the case that will reproduce this issue with the command you used?

**Comment by HassanKhadour** [comment]
Can one of the admins verify this patch?

**Comment by galpress** [comment]
@firasj can you please provide the command lines we were running?
@sshaulnv what is needed to get this merged? Do you want me to test this on mlx5 hardware?

**Comment by HassanKhadour** [comment]
Hi Gal, yes we need the commands in order to test it on mlx5 hardware

**Comment by firasj** [comment]
@gal-pressman Can you please close this PR for now?
It's been on a low priority and we'll reopen once we have a reproduction and an answer.

**Comment by mrgolin** [comment]
Hi, running:
```
for i in {1..20}; do ib_send_lat -c SRD -x 0 -F --report_gbits -a -n 10000 -r 1 -e & ib_send_lat -c SRD -x 0 -F --report_gbits -a -n 10000 -r 1 -e localhost; done
```
is reproducing the issue on EFA HW.
@sshaulnv @gal-pressman can you please test this with mlx5 HW?

File: src/raw_ethernet_resources.c [hunk 1]
```diff
@@ -1005,8 +1005,7 @@ int run_iter_fw(struct pingpong_context *ctx,struct perftest_parameters *user_pa
 		}
 
 		if (user_param->use_event) {
-
-			if (ctx_notify_events(ctx->channel)) {
+			if (ctx_notify_send_recv_events(ctx)) {
```
Comment (sshaulnv) [inline src/raw_ethernet_resources.c:1008]: @gal-pressman, can you describe this change please?

**Comment by mrgolin** [comment]
@HassanKhadour Can you please merge this?

**Comment by HassanKhadour** [comment]
Hi @mrgolin , conflicts should be resolved to merge.

**Comment by mrgolin** [comment]
@HassanKhadour I've rebased and pushed as new PR https://github.com/linux-rdma/perftest/pull/211.

**Pull request closed without merging**